extmod/modjson.c

```diff
@@ -336,6 +336,7 @@ static mp_obj_t mod_json_load(mp_obj_t stream_obj) {
             }
         }
     }
+    goto fail;
```

Comment:
This is very neat!  A simple change, very low cost.

But, I can't help but think it would be better to put this `goto` up on line 163, with a comment about what's happening:
```c
    ...
    for (;;) {
    cont:
        if (S_END(s)) {
            // Input finished abruptly in the middle of a composite entity.
            goto fail;
        }
        mp_obj_t next = MP_OBJ_NULL;
        ...
```

That should be equivalent, and easier to reason about.

Reply:
The rationale for choosing that location in my commit was to separate the failure action from the main processing loop - as in, "the whole input buffer was consumed, execute the next step in the process", otherwise it might have gotten lost in a fairly visually dense bit of code.

I agree a comment should have been there to begin with though :)

Still, your request is reasonable and indeed equivalent code-wise, I've updated the PR to reflect those changes.